File: minesweeper/java-minesweeper/src/main/java/it/chicio/minesweeper/field/parser/FieldsParserByDivingInputIntoRows.java
package it.chicio.minesweeper.field.parser;

import it.chicio.minesweeper.field.Field;

import java.util.ArrayList;
import java.util.List;

public class FieldsParserByDivingInputIntoRows implements FieldsParser {
    private FieldRowParser fieldRowParser;

    public FieldsParserByDivingInputIntoRows(FieldRowParser fieldRowParser) {
        this.fieldRowParser = fieldRowParser;
    }

    public List<Field> parse(String fieldsAsString) throws RuntimeException {
        FieldsParsingStatus fieldsParsingStatus = new FieldsParsingStatus(null, null, 0, 0, new ArrayList<Field>());
        for (String row : fieldsAsString.split(System.getProperty("line.separator"))) {
            fieldsParsingStatus = fieldRowParser.parse(row, fieldsParsingStatus);
        }
        return fieldsParsingStatus.fieldsParsed;
    }
}

File: minesweeper/kotlin-mockk-minesweeper/src/main/kotlin/it/chicio/minesweeper/field/parser/FieldsParserByDivingInputIntoRows.kt
package it.chicio.minesweeper.field.parser

import it.chicio.minesweeper.field.Field

class FieldsParserByDivingInputIntoRows(private val fieldRowParser: FieldRowParser) : FieldsParser {
    override fun parse(fields: String): List<Field> {
        var fieldsParsingStatus = FieldsParsingStatus(null, null, 0, 0, mutableListOf())
        fields
                .split(System.getProperty("line.separator"))
                .toTypedArray()
                .forEach { row ->
                    fieldsParsingStatus = fieldRowParser.parse(row, fieldsParsingStatus)
                }
        return fieldsParsingStatus.fieldsParsed
    }
}